src/backends/plonky2/primitives/merkletree_circuit.rs

```diff
@@ -21,39 +21,43 @@ use plonky2::{
     },
     plonk::circuit_builder::CircuitBuilder,
 };
+use std::iter;
 
 use crate::backends::plonky2::basetypes::{Hash, Value, D, EMPTY_HASH, EMPTY_VALUE, F, VALUE_SIZE};
+use crate::backends::plonky2::common::{
+    CircuitBuilderPod, OperationTarget, StatementTarget, ValueTarget,
+};
 use crate::backends::plonky2::primitives::merkletree::MerkleProof;
 
 /// `MerkleProofCircuit` allows to verify both proofs of existence and proofs
 /// non-existence with the same circuit.
 /// If only proofs of existence are needed, use `MerkleProofExistenceCircuit`,
 /// which requires less amount of constraints.
 pub struct MerkleProofCircuit<const MAX_DEPTH: usize> {
-    root: HashOutTarget,
-    key: Vec<Target>,
-    value: Vec<Target>,
-    existence: BoolTarget,
-    siblings: Vec<HashOutTarget>,
-    case_ii_selector: BoolTarget, // for case ii)
-    other_key: Vec<Target>,
-    other_value: Vec<Target>,
+    pub root: HashOutTarget,
+    pub key: ValueTarget,
```

Comment:
Nice! we discussed this with @ax0 a couple of days ago, about adding this `ValueTarget` type, but seemed a bit extra just for the tree circuit, but makes a lot of sense to have it since it is used at the other parts aside from the tree circuit.